api/src/main/java/marquez/api/OpenLineageResource.java

```diff
@@ -68,6 +68,7 @@ public OpenLineageResource(
   public void create(@Valid @NotNull BaseEvent event, @Suspended final AsyncResponse asyncResponse)
       throws JsonProcessingException, SQLException {
     if (event instanceof LineageEvent) {
+      serviceFactory.getSearchService().indexEvent((LineageEvent) event);
```

Comment:
If `search.enabled=false`, will the index call fail/error?

Reply:
Handling in the `indexEvent` method:
```java
  public void indexEvent(@Valid @NotNull LineageEvent event) {
    if (!searchConfig.isEnabled()) {
      log.debug("Search is disabled, skipping indexing");
      return;
    }
    UUID runUuid = runUuidFromEvent(event.getRun());
    log.debug("Indexing event {}", event);

    if (event.getInputs() != null) {
      indexDatasets(event.getInputs(), runUuid, event);
    }
    if (event.getOutputs() != null) {
      indexDatasets(event.getOutputs(), runUuid, event);
    }
    indexJob(runUuid, event);
  }
```

Reply:
Ok, when we follow up with an SearchService **interface**, we'll want to bind search to an engine (`psql`, `opensearch`) and can do away with the flag.